plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/refactoring/wizards/rename/PyRenameGlobalProcess.java

```diff
@@ -37,18 +46,104 @@ protected boolean getRecheckWhereDefinitionWasFound() {
+                            foundDefs,
+                            occurrence.endLine, occurrence.endCol, module.getNature(), completionCache, false);
 
+                    for (IDefinition def : foundDefs) {
+                        IModule defModule = def.getModule();
+                        if (defModule == null || (def instanceof Definition && ((Definition) def).ast == null)) {
+                            continue;
+                        }
+                        for (IDefinition actualDef : actualDefinitions) {
+                            IModule actualDefModule = actualDef.getModule();
+                            if (defModule.getName().equals(actualDefModule.getName())
+                                    && def.getLine() == actualDef.getLine()) {
+                                ret.add(occurrence);
+                            } else if (checkDef(def, module, completionCache, actualDef, false)) {
+                                ret.add(occurrence);
+                            }
+                        }
+                    }
+
+                }
+            } catch (Exception e) {
+                Log.log(e);
+            }
+        }
         return ret;
     }
 
+    private boolean checkDef(IDefinition iDef, SourceModule module, ICompletionCache completionCache,
+            IDefinition iActualDef, boolean checkModule)
+            throws CompletionRecursionException, Exception {
+        if (iDef == null) {
+            return false;
+        }
+        if (iDef instanceof Definition) {
+            if (checkModule) {
+                if (((Definition) iDef).module == null || ((Definition) iDef).ast == null) {
+                    return false;
+                }
+                String defModuleName = ((Definition) iDef).module.getName();
+                if (iActualDef.getModule().getName().equals(defModuleName)
+                        && iDef.getLine() == iActualDef.getLine()) {
+                    return true;
+                }
+            }
+            List<IDefinition> foundDefs = new ArrayList<IDefinition>();
+            if (iDef instanceof AssignDefinition) {
+                AssignDefinition assignDef = (AssignDefinition) iDef;
+                PyRefactoringFindDefinition.findActualDefinition(request.getMonitor(), module,
+                        assignDef.target,
+                        foundDefs,
+                        assignDef.line, assignDef.col, module.getNature(), completionCache,
+                        false);
+                PyRefactoringFindDefinition.findActualDefinition(request.getMonitor(), module,
+                        assignDef.value,
+                        foundDefs,
+                        assignDef.line, assignDef.col, module.getNature(), completionCache,
+                        false);
+                PyRefactoringFindDefinition.findActualDefinition(request.getMonitor(), module,
+                        assignDef.type,
```

Comment:
I think you should check if the target/value/type are non-null and non-empty before proceeding to check those. Also, since the definition has the actual nodes for the line/col of the type and value, I think you should use those instead of the `assignDef.line` and `assignDef.col` (so, you can use the `assignDef.nodeType.beginColumn/assignDef.nodeType.beginLine` and likewise for `assignDef.nodeValue` -- the target can use the assignDef.line and assignDef.col as it should be the same thing).

Make sure you also check if the nodes aren't null before accessing them.

Reply:
You mean check for `assignDef.nodeType.beginColumn/assignDef.nodeType.beginLine` and `assignDef.nodeValue.beginColumn/assignDef.nodeValue.beginLine` instead of assignDef.type and assignDef.value? And so should the check for assignDef.target remain as it is?

Reply:
Yes to all (assignDef.target maps to the assignDef.line/assignDef.col, but assignDef.value and assignDef.type don't).

Reply:
Updated in last commits ;)